lib/k8s-e2e-tests/tests/vector-agent.rs

```diff
@@ -218,7 +230,16 @@ async fn simple_custom_config() -> Result<(), Box<dyn std::error::Error>> {
         )
         .await?;
 
-    let test_namespace = framework.namespace(&pod_namespace).await?;
+    let test_namespace = framework
+        .namespace(namespace::Config::from_namespace(&Namespace {
+            metadata: ObjectMeta {
+                name: Some(pod_namespace.clone()),
+                ..Default::default()
+            },
+            spec: None,
+            status: None,
+        })?)
```

Comment:
I think it would be nice if this could be factored out into a separate function since every test uses it.

```rust
fn labelled_namespace_or_some_other_better_name(pod_namespace: &str) -> Result<WhateverItIs> {
    Namespace {
            metadata: ObjectMeta {
                name: Some(pod_namespace.clone()),
                ..Default::default()
            },
            spec: None,
            status: None,
        })
}
```

Reply:
Think extract into the `namespace` module? There's something similar for pods, but has a whole separate file which I don't see the point of.

Reply:
Definitely don't disagree with adding a helper for this - was just intended to get working tests for the feature :)